root/static/scripts/tests/Control/URLCleanup.js

```diff
@@ -1686,6 +1686,13 @@ const testData = [
             expected_clean_url: 'https://www.instagram.com/nathanwpylestrangeplanet/',
        only_valid_entity_types: ['artist', 'event', 'label', 'place', 'series'],
   },
+  {
+                     input_url: 'https://www.instagram.com/stories/nathanwpylestrangeplanet/',
+             input_entity_type: 'recording',
+    expected_relationship_type: 'streamingfree',
+            expected_clean_url: 'https://www.instagram.com/nathanwpylestrangeplanet/',
+       only_valid_entity_types: ['release'],
+  },
```

Comment:
Did we allow linking these to releases before? If not, is there a reason we should allow it now?

Reply:
Yes we did/do allow it. Actually we neither match nor validate it for release (and area, instrument, release group, work). This patch doesn’t change anything to the release editor regarding Instagram URLs.

Reply:
Oh, so basically we didn't (and don't) autoselect but we don't specifically block it if people hand-pick it? In this specific case I'm not sure it makes sense to link to these, we might want to specifically disallow them. After all, a profile page for an album or whatnot would be RG level, not release level. If anything, a streaming page for a release could sometimes make sense but only as /p/ (if there's a video containing the entire release).

Reply:
To block it, it has to match a relationship type. We could easily do it in release editor since there is a `socialnetwork.release` entry in `LINK_TYPES` but we cannot do it for other entity types without a bit of refactoring.

There currently are:

* 58 K Instagram URLs in the database.

* 18 /p/ Instagram URLs linked to MB releases:
   * `https://www.instagram.com/p/B_7IG9gonk0/` [[info](https://musicbrainz.org/url/7680e1b6-efe1-4b6e-bd04-f7178d7ddba2)]
    * `https://www.instagram.com/p/BccvnTqgtON/` [[info](https://musicbrainz.org/url/551e8019-0500-4b70-85c0-3840aa7f7e50)]
    * `https://www.instagram.com/p/Be-fnmVgHYC/` [[info](https://musicbrainz.org/url/0f662302-6c35-4bc3-83c7-36a16756999c)]
    * `https://www.instagram.com/p/Be-Pht-ge3H/` [[info](https://musicbrainz.org/url/5e18a87b-59d9-42a4-8279-b71da86f6a05)]
    * `https://www.instagram.com/p/Be-QBCig3Sw/` [[info](https://musicbrainz.org/url/772644fb-218f-4d25-89ec-017ec4fb438c)]
    * `https://www.instagram.com/p/Be-QQR2AlQu/` [[info](https://musicbrainz.org/url/2aa5c500-2bfd-414f-96e1-24b51881d749)]
    * `https://www.instagram.com/p/Be-R4oAgY50/` [[info](https://musicbrainz.org/url/b1d08c85-0b06-4bc0-b1df-c90e5c2c7b38)]
    * `https://www.instagram.com/p/Be-Rr2lgqLI/` [[info](https://musicbrainz.org/url/92fefa58-0705-4660-9bd3-db47c90631f1)]
    * `https://www.instagram.com/p/Be-S2-7A84P/` [[info](https://musicbrainz.org/url/ba4a8e77-ca33-434b-af64-438e5047ff5d)]
    * `https://www.instagram.com/p/BesgjIKAqNh/` [[info](https://musicbrainz.org/url/97ad41e6-5613-4a41-addf-6a74ba15b1c8)]
    * `https://www.instagram.com/p/BesgLByg03V/` [[info](https://musicbrainz.org/url/dc92f333-925f-4a55-88c6-66d5aea8a840)]
    * `https://www.instagram.com/p/Be-SGrLAiE5/` [[info](https://musicbrainz.org/url/8b6109b9-54d8-44e1-9764-c7134c1659a8)]
    * `https://www.instagram.com/p/Be-SkzPgbh_/` [[info](https://musicbrainz.org/url/b80649ab-b33d-4961-b134-28b5b05d9f3f)]
    * `https://www.instagram.com/p/Be-TF3yA3gM/` [[info](https://musicbrainz.org/url/bec34482-0729-4958-accf-35a17bb03a45)]
    * `https://www.instagram.com/p/BetIBp7AYgJ/` [[info](https://musicbrainz.org/url/240e971d-5291-41d5-82dc-e67196e3b0c8)]
    * `https://www.instagram.com/p/BhCeOgmFi58/` [[info](https://musicbrainz.org/url/0cae706e-aeaf-4b0e-a3df-9d3915bee9a4)]
    * `https://www.instagram.com/p/BjHz-y0hrbA/` [[info](https://musicbrainz.org/url/2475e928-50f4-4ffc-ba10-559feb44f548)]
    * `https://www.instagram.com/p/BpXEVRahfko/` [[info](https://musicbrainz.org/url/b83bf993-b5c0-45dc-9d21-6d7a2bbe7191)]

* 2 non /p/ Instagram URLs linked to releases:
    * `https://www.instagram.com/bastardix_/` [[info](https://musicbrainz.org/url/d29f879d-0a39-45f6-8640-24a3a07237b0)]
    * `https://www.instagram.com/standingsunmusic/` [[info](https://musicbrainz.org/url/55a1d499-1bab-4652-b288-eac23f97f649)]

* 0 Instagram URL linked to either release group or work.

It seems unneeded to handle release groups and works. I would be inclined to handle releases the same as standalone recordings, and to suggest adding release as well. As a STYLE BDFL please decide that should be changed.

Reply:
Sigh, the /p/ links seem to mostly be cover art. Wonder if we need to show a message about not linking cover art... but that'd be a non-error message and we don't even support that, so probably not now. The two profile URLs shouldn't be linked to a release anyway, and are neither a streaming page nor a discography entry. The safest thing seems to be to autoselect free stream for release, block non /p/ and hope for the best, and ignore other entities for now (so, what you suggested, I think? :) )

Reply:
Changed through https://github.com/metabrainz/musicbrainz-server/pull/1590/commits/aa44302666579eca00564c7a4534616d86ec8309 and https://github.com/metabrainz/musicbrainz-server/pull/1590/commits/7787de7e643c9c3cd25466e084cbdf323c5a7c71.